### PR TITLE
MAINT: Use inst_id, not sat_id for instruments

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #473 

Replaces `sat_id` with `inst_id` everywhere.

## Type of change

- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

```
import pysat
test_inst = pysat.Instrument('pysat','testing', inst_id='100')  
test_inst.load(2009, 1)
print(test_inst)
```
Properties should pass through accordingly, load 100 data points.

**Test Configuration**:
* Mac OS X 10.15.3
* python 3.7.3

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
NOTE: this will require changes in all downstream packages that include instruments.
- [ ] pysatCDAAC 
- [ ] pysatNASA 
- [ ] pysatIncubator
- [ ] pysatMadrigal 
- [ ] pysatMissions
- [ ] pysatModels
- [ ] pysatSpaceWeather

**Comment by jklenzing** [comment]
Pull requests at all downstream packages.  Space Weather and Missions still have some unrelated testing issues, but everything else works locally.

**Review by rstoneback — APPROVED** [review]
Thanks Jeff. Looks good to me.

**Comment by jklenzing** [comment]
Thanks @rstoneback.  Still tidying things up from the last round of downstream pulls.  @aburrell, is this one up to standards?

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
   - Add tests for acknowledgements and references
   - Removed implicit conversion to integers in
     methods.general.convert_timestamp_to_datetime
+  - Rename sat_id as inst_id
```
Comment (aburrell) [inline CHANGELOG.md:67]:
```suggestion
  - Renamed `sat_id` Instrument keyword argument to `inst_id`
```

File: pysat/_constellation.py [hunk 2]
```diff
@@ -70,11 +70,11 @@ def __str__(self):
         ninst = len(self.instruments)
 
         if ninst > 0:
-            output_str += "\nIndex Platform Name Tag Sat_ID\n"
+            output_str += "\nIndex Platform Name Tag Inst_ID\n"
             output_str += "------------------------------\n"
```
Comment (aburrell) [inline pysat/_constellation.py:74]:
```suggestion
            output_str += "-------------------------------\n"
```

**Review by aburrell — APPROVED** [review]

**Pull request merged**